### PR TITLE
fix flaky 'restart triggers after shutdown' test

### LINKED ISSUE #6921: Trigger service persistence tests are flaky
Currently the trigger service tests for persistence have been disabled since they are flaky. We should find the cause and reenable them https://github.com/digital-asset/daml/blob/39b6e47a7e5dda6d5aad71cfca47d8d2df1722d6/triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceTest.scala#L509

### PR DESCRIPTION
**Pull request by S11001001**
Fixes #6921 by ensuring the service is actually shut down before continuing with the restart.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by S11001001** [comment]
At first glance, since #6927 had me thinking about the function, I would say this is because the change at https://github.com/digital-asset/daml/pull/4593#discussion_r384555266 needs to be made in `withTriggerService`, too. Otherwise, the behavior expected by this comment won't happen reliably:

https://github.com/digital-asset/daml/blob/39b6e47a7e5dda6d5aad71cfca47d8d2df1722d6/triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceTest.scala#L520-L521

Just as the problem fixed by #6927 was seen locally because things were happening too fast, this happens on CI (but not locally) because things are happening too slowly.

File: triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala [hunk 1]
```diff
@@ -163,14 +163,21 @@ object TriggerServiceFixture {
       a <- testFn(uri, client, ledgerProxy)
     } yield a
 
-    fa.onComplete { _ =>
```
Comment (rohanjr) [inline triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:166]: Ah, so the problem was that the shutdown here would race with the second use of `withTriggerService` in the test?

Comment (S11001001) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:137]: Yes.

**Comment by S11001001** [comment]
d5dd8bf did the trick (CI failed for unrelated reason, the tests in question passed though).

**Comment by S11001001** [comment]
"restart triggers with update errors" seems like it might be flaky, too; worth examining in separate issue/PR.

File: triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala [hunk 2]
```diff
@@ -134,15 +136,28 @@ object TriggerServiceFixture {
...
-      serviceF.foreach({ case (_, system) => system ! Stop })
-      ledgerF.foreach(_._1.close())
-      toxiproxyF.foreach(_._1.destroy)
+    fa.transformWith { ta =>
+      for {
```
Comment (cocreature) [inline triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: What exactly went wrong with `Future.sequence`? As far as I understand `Future.sequence` is not fail fast so it shouldn’t stop anything from running. Does running these things in parallel cause issues?

Comment (cocreature) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: Actually, I think I might have misread something and it is fail fast which probably explains the issue here.

Comment (SamirTalwar-DA) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: I don't think `Future.sequence` is fail-fast. It's passed already-running Futures and AFAIK, there's no way to stop them.

Comment (cocreature) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: So the issue was that we tried to stop things in parallel? I’m not quite sure why that’s a problem.

Comment (S11001001) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: It is fail-fast in the sense that if one future fails, it completed, allowing the next step (the restart) to proceed despite something else's shutdown being in progress.

Comment (S11001001) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: I could possibly restore the parallel shutdown, if the forced success of this chain is really the relevant piece, but would rather get this in as is first.

Comment (cocreature) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala:140]: Makes sense, thanks for the explanation! I’m not worried about losing parallelism so totally fine to get this in and ignore this.

**Pull request merged**